x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_modal.tsx

```diff
@@ -0,0 +1,270 @@
+   */
+  setIsModalVisible: (isModalVisible: boolean) => void;
+}
+
+/**
+ * Modal for editing the highlighted fields of a rule.
+ */
+export const HighlighedFieldsModal: FC<HighlighedFieldsModalProps> = ({
+  fieldOptions,
+  rule,
+  customHighlightedFields,
+  dataFormattedForFieldBrowser,
+  setIsEditLoading,
+  setIsModalVisible,
+}) => {
+  const { addSuccess } = useAppToasts();
+  const { euiTheme } = useEuiTheme();
+  const { mutateAsync: updateRule } = useUpdateRule();
+  const modalTitleId = useGeneratedHtmlId();
+
+  const defaultFields = useHighlightedFields({
+    dataFormattedForFieldBrowser,
+    investigationFields: customHighlightedFields,
+    type: 'default',
+  });
+  const defaultFieldsArray = useMemo(() => Object.keys(defaultFields), [defaultFields]);
+
+  const customFields = useMemo(
+    () => customHighlightedFields.map((field: string) => ({ label: field })),
+    [customHighlightedFields]
+  );
+
+  const [selectedOptions, setSelectedOptions] = useState(customFields);
+
+  const options = useMemo(
+    () => fieldOptions.map((field) => ({ label: field.name })),
+    [fieldOptions]
+  );
+
+  const { form } = useForm({
+    defaultValue: { customHighlightedFields: [] },
+    schema,
+  });
+
+  const onCancel = useCallback(() => {
+    setIsModalVisible(false);
+  }, [setIsModalVisible]);
+
+  const onSubmit = useCallback(async () => {
+    setIsEditLoading(true);
+
+    const updatedRule = await updateRule({
+      ...rule,
+      id: undefined,
+      investigation_fields:
+        selectedOptions.length > 0
+          ? { field_names: selectedOptions.map((option) => option.label) }
+          : undefined,
+    } as RuleUpdateProps);
+
+    addSuccess(SUCCESSFULLY_SAVED_RULE(updatedRule?.name ?? 'rule'));
+    setIsEditLoading(false);
+    setIsModalVisible(false);
+  }, [updateRule, addSuccess, rule, setIsModalVisible, setIsEditLoading, selectedOptions]);
+
+  return (
+    <EuiModal
+      css={{ width: '600px' }}
+      data-test-subj={HIGHLIGHTED_FIELDS_MODAL_TEST_ID}
+      id={modalTitleId}
+      onClose={onCancel}
+    >
+      <EuiModalHeader data-test-subj={HIGHLIGHTED_FIELDS_MODAL_TITLE_TEST_ID}>
+        <EuiModalHeaderTitle id={modalTitleId} size="s">
+          <FormattedMessage
+            id="xpack.securitySolution.flyout.right.investigation.highlightedFields.modalTitle"
+            defaultMessage="Edit highlighted fields"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiText size="s" data-test-subj={HIGHLIGHTED_FIELDS_MODAL_DESCRIPTION_TEST_ID}>
+          <FormattedMessage
+            id="xpack.securitySolution.flyout.right.investigation.highlightedFields.modalDescription"
+            defaultMessage="Changes made here will be applied to the {ruleName} rule. Any custom fields you add will be displayed in all alerts generated by this rule."
+            values={{ ruleName: <b>{rule.name}</b> }}
```

Comment:
Nice touch to bold the rule name. When you were presenting your WIP work earlier I felt it was missing. This looks much better!